apps/src/netsim/netsimUtils.js

```diff
@@ -207,6 +207,28 @@ exports.deserializeNumber = function (storedNum) {
 };
 
 /**
+ * Helper for converting from an older header-spec format to a new, simpler one.
+ * Old format: {key:{string}, bits:{number}}[]
+ * New format: string[]
+ * If we detect the old format, we return a spec in the new format.
+ * @param {Array} spec
+ * @returns {Array}
+ */
+exports.scrubHeaderSpecForBackwardsCompatibility = function (spec) {
+  var scrubbedSpec = [];
+  spec.forEach(function (specEntry) {
+    if (typeof specEntry === 'string') {
+      // This is new new format, we can just copy it over.
+      scrubbedSpec.push(specEntry);
+    } else if (specEntry !== null && typeof specEntry === 'object') {
+      // This is the old {key:'', bits:0} format.  We just want the key.
+      scrubbedSpec.push(specEntry.key);
```

Comment:
Good idea.  I have a similar set of warnings for a different issue in `scrubLevelConfiguration` below.  I'll implement that here.